src/main/java/org/snomed/snowstorm/core/util/SPathUtil.java

```diff
@@ -0,0 +1,41 @@
+package org.snomed.snowstorm.core.util;
+
+import io.kaicode.elasticvc.api.PathUtil;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+// Snowstorm PathUtil has domain specific concepts irrelevant to the ElasticVC library.
+public class SPathUtil {
+    /**
+     * Return the given Branch path's ancestors. The given Branch path is not included in the ancestors,
+     * and processing stops at the CodeSystem level, i.e. processing an Extension's Branch path will stop at MAIN/SNOMEDCT-XX.
+     *
+     * @param branchPath Find ancestors for this Branch path.
+     * @return The given Branch path's ancestors.
+     */
+    public static List<String> getAncestors(String branchPath) {
+        if (branchPath == null || branchPath.isEmpty() || isCodeSystemBranch(branchPath) || !isExpectedFormat(branchPath)) {
```

Comment:
It would be better to throw IllegalArgumentException here when branchPath is Null or empty or not starting with MAIN.

Reply:
Thanks @CoderMChu, I have made that change.